com.ibm.streamsx.utility/com.ibm.streamsx.utility/native.function/utility.cpp

```diff
@@ -104,4 +105,10 @@ namespace com { namespace ibm { namespace streamsx { namespace utility {
         if (rc<0) THROW (SPLRuntimeOperator, "could not set processor affinity to CPUs"
             << ", " << strerror(errno));
     }
+
+    void setThreadName(SPL::rstring name)
+    {
+        pthread_setname_np(pthread_self(), name.c_str());
```

Comment:
Ah, yeah.  Sorry about that.  For error checking, I was thinking just trimming the name to 15 real characters if it's too long (which is what the prctl(PR_SET_NAME) would do), and then also check and THROW for other errors (in case something oddball pops out of the call, similar to what the setCPUAffinity functions about do.

Any complaints?

For the RHEL6 issue, I'm not able to reproduce the failure on RHEL6, with some test code.  At least RHEL6.6 and later all seem to work, provided we link with libpthread (which I overlooked and was getting for free in my actual use cases from the larger application already linking it).  I will add the dependency to the Makefile when we generate the .so

Was there a particular version of RHEL6 that you were seeing this problem on?